src/createEpicMiddleware.ts

```diff
@@ -106,8 +122,29 @@ export function createEpicMiddleware<
         'epicMiddleware.run(rootEpic) called before the middleware has been setup by redux. Provide the epicMiddleware instance to createStore() first.'
       );
     }
+
+    rootEpic.__RO__SIGTERM_ID__ = Math.random().toString(16);
     epic$.next(rootEpic);
   };
 
+  epicMiddleware.sigterm = (epic) => {
+    // sends sigterm to the given epic
+    store.dispatch({ type: `${SIGTERM}/${epic.__RO__SIGTERM_ID__}` });
+  };
+
+  epicMiddleware.sigtermAll = (epic) => {
+    // sends sigterm to all currently running epics
+    store.dispatch({ type: `${SIGTERM}/all` });
+  };
+
+  epicMiddleware.kill = rootEpic => {
+    // this should only kill the given epic, alias of `complete`.
+  };
+
+  epicMiddleware.killall = () => {
+    // running epics could be stored into a list when calling `middleware.run`
+    // this method could flush that list and call `complete`
+  };
```

Comment:
this would address give the ability of killing/terminating one or all the currently running epics

Reply:
Implementation details:

1. assign an `ID` to every epic passed to `middleware.run`
2. store every epic passed to `middleware.run` into a long-living variable
3. use the ID to terminate a specific epic
4. use the list of running epic to terminate everything indiscriminately